src/mcp/server/fastmcp/tools/base.py

```diff
@@ -88,13 +94,20 @@ async def run(
     ) -> Any:
         """Run the tool with arguments."""
         try:
-            return await self.fn_metadata.call_fn_with_arg_validation(
+            result = await self.fn_metadata.call_fn_with_arg_validation(
                 self.fn,
                 self.is_async,
                 arguments,
                 {self.context_kwarg: context}
                 if self.context_kwarg is not None
                 else None,
             )
+            if self.output and self.output.get("type") == "object":
```

Comment:
This logic is debatable it introduces a breaking change, potential to make this opt-in or opt-out by adding a flag to the @mcp.tool() annotation something like @mcp.tool(enable_data_content=True)